cloudsmith/data_source_package_download_test.go

```diff
@@ -0,0 +1,196 @@
+package cloudsmith
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccPackageDownload_data(t *testing.T) {
+	t.Parallel()
+
+	namespace := os.Getenv("CLOUDSMITH_NAMESPACE")
+	apiKey := os.Getenv("CLOUDSMITH_API_KEY")
+	userName := "token"
+	repoName := "terraform-download-test"
+	repoConfig := testAccDataSourcePackageDownload_repositoryConfig(repoName, namespace)
+	fileName := "hello.txt"
+	packageName := "hello"
+	packageVersion := "1.0"
+	destinationPath := "."
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: repoConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("cloudsmith_repository.test", "name", repoName),
+					resource.TestCheckResourceAttr("cloudsmith_repository.test", "namespace", namespace),
+				),
+			},
+			{
+				PreConfig: func() {
+					createHelloWorldFile(fileName)
+					uploadPackageToCloudsmith(apiKey, namespace, repoName, fileName, packageName, packageVersion, userName)
+				},
+				Config: testAccPackageDownloadData(namespace, repoName, packageName, packageVersion, destinationPath),
```

Comment:
i think this is going to destroy your repository as you've removed it from the config here, that doesn't seem like what you want